doc/specs/#1012 - Show dependencies.md

```diff
@@ -0,0 +1,124 @@
+
+### install
+The install command will enumerate the dependencies of the package version being installed as follows:
+```
+> winget install Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsoft.WindowsTerminal
+Downloading https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+Successfully verified installer hash
+Starting package install...
+```
+
+### show
+The show command will enumerate the dependencies of the package as follows:
+```
+> winget show Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Version: 7.9.5
+Publisher: Notepad++ Team
+Author: Don Ho
+Moniker: notepad++
+Description: Notepad++ is a free (as in “free speech” and also as in “free beer”) source code editor and Notepad replacement that supports several languages. Running in the MS Windows environment, its use is governed by GNU General Public License.
+Homepage: https://notepad-plus-plus.org/
+License: GPL-2.0-only
+License Url: https://raw.githubusercontent.com/notepad-plus-plus/notepad-plus-plus/v7.9.5/LICENSE
+Installer:
+  Type: Nullsoft
+  Locale: en-US
+  Download Url: https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+  SHA256: 4881548cd86491b453520e83c19292c93b9c6ce485a1f9eb9301e3913a9baced
+  Dependencies:
+    - Windows Feature: Hyper-V
+    - Package: Microsoft.WindowsTerminal
+```
+
+### upgrade
+The upgrade command will enumerate the dependencies of the package as follows:
+```
+> winget upgrade Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsoft.WindowsTerminal
+Successfully verified installer hash
+Starting package install...
+```
+As of now, it will not try to validate nor install any of the dependencies for any package version.
+
+### uninstall
+```
+> winget uninstall Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Package had dependencies that may not be needed anymore:
+  - Windows Feature: Hyper-V
+  - Package: Microsoft.WindowsTerminal
+Starting package uninstall...
+Successfully uninstalled
+```
+Dependencies will not be uninstalled.
+
+### validate
+Will show dependencies for all of the installers found.
+```
+Manifest has the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsoft.WindowsTerminal
+Manifest validation succeeded.
+```
+
+### import
+Will gather all the dependencies and show them together before starting.
+```
+The packages found in this import have the following dependencies:
+  - Windows Feature: Hyper-V, Containers

```

Comment:
I'm not a huge fan of this model of having them all on one line, but I can see the value for import.

Also, you need to have examples of at least External dependencies to see how they fit into this model.  I don't think that putting them on one line makes sense.

Reply:
Ok, something like a nested bullet point will be better?

Reply:
No, I think that a whole line for the type is wasted.  I would rather have each on its own line, or multiple on the line (like you had it).  I wouldn't expect many packages to have loads of dependencies, although a graph could explode.

Own line option:
```
Windows Feature: Hyper-V
Windows Feature: Containers
Package: Foo.Foo
```